app/routes/formbuilder/edit.js

```diff
@@ -12,17 +12,41 @@ export default class FormbuilderEditRoute extends Route {
     this.registerCustomFields();
   }
 
-  model(params) {
-    return this.store.findRecord('generated-form', params.id);
+  async model(params) {
+    return await this.getGeneratedFormById(params.id);
   }
 
   setupController(controller, model) {
     super.setupController(...arguments);
     controller.refresh.perform({
-      value: model.ttlCode ? model.ttlCode : template,
+      formTtlCode: this.getFormTtlCode(model),
+      resetBuilder: false,
+      isInitialRouteCall: true,
     });
   }
 
+  resetController(controller) {
+    if (
+      controller.builderStore &&
+      controller.builderStore.observers[controller.REGISTERED_FORM_TTL_CODE_KEY]
```

Comment:
I would consider `.observers` a private implementation detail of the class. `deregisterObserver` also doesn't fail if the key was already removed, so I would just always call it.